core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -473,7 +473,11 @@ public Tenant getTenant(int tenantId) throws UserStoreException {
                 tenant.setRealmConfig(realmConfig);
                 setSecondaryUserStoreConfig(realmConfig, tenantId);
                 tenant.setAdminName(realmConfig.getAdminUserName());
-                tenant.setAdminUserId(realmConfig.getAdminUserId());
+                if (tenant.getAssociatedOrganizationUUID() != null) {
+                    tenant.setAdminUserId(resolveTenantAdminUserId(realmConfig, tenant));
+                } else {
+                    tenant.setAdminUserId(realmConfig.getAdminUserId());
```

Comment:
user id was added to realmConfig recently. The tenant admin's user id won't be available in the realm config for older tenants. If the realm config doesn't contain the value for tenant admin need to resolve it using the user name similar to `getUserId(tenant.getAdminName(), tenant.getId())` in `getTenant(String tenantUniqueID)`  method

Reply:
The userID resolving logic is refactored accordingly
https://github.com/wso2/carbon-kernel/pull/3529/files#diff-d39d58208931ad0eacf4bd33808d5efa80a5e1851d969deecdceaa1d6dd264cfR1523-R1538